include/exiv2/config.h

```diff
@@ -93,5 +93,15 @@ typedef int pid_t;
 #endif
 //////////////////////////////////////
 
+// https://softwareengineering.stackexchange.com/questions/291141/how-to-handle-design-changes-for-auto-ptr-deprecation-in-c11
+#if __cplusplus >= 201103L
+  #include <memory>
+  #include <sys/types.h>
+  #include <unistd.h>
+  template <typename T>
+  using auto_ptr = std::unique_ptr<T>;
+#else
```

Comment:
That's a clever trick, I did not think about that. 

I think you can remove lines 103 & 104 from here. We do not need the `using` instruction when not using C++11.

Reply:
I can't remember what `using` does.  I got the code from StackOverflow (in the comment).